src/material/datepicker/date-range-input.spec.ts

```diff
@@ -865,6 +865,45 @@ describe('MatDateRangeInput', () => {
     expect(endInput.errorStateMatcher).toBe(matcher);
   });
 
+
+  it('should only update model for input that changed', fakeAsync(() => {
+    const fixture = createComponent(RangePickerNgModel);
+
+    fixture.detectChanges();
+    tick();
+
+    expect(fixture.componentInstance.startDateModelChangeCount).toBe(0);
+    expect(fixture.componentInstance.endDateModelChangeCount).toBe(0);
+
+    fixture.componentInstance.rangePicker.open();
+    fixture.detectChanges();
+    tick();
+
+    const fromDate = new Date(2020, 0, 1);
+    const toDate = new Date(2020, 0, 2);
+    fixture.componentInstance.rangePicker.select(fromDate);
+    fixture.componentInstance.rangePicker.select(toDate);
+    fixture.detectChanges();
+    tick();
+
+    expect(fixture.componentInstance.startDateModelChangeCount).toBe(1, 'Start Date 1');
+    expect(fixture.componentInstance.endDateModelChangeCount).toBe(1, 'End date 1');
+
+    fixture.componentInstance.rangePicker.open();
+    fixture.detectChanges();
+    tick();
+
+    const fromDate2 = new Date(2021, 0, 1);
+    const toDate2 = new Date(2021, 0, 2);
+    fixture.componentInstance.rangePicker.select(fromDate2);
+    fixture.componentInstance.rangePicker.select(toDate2);
+    fixture.detectChanges();
+    tick();
+
+    expect(fixture.componentInstance.startDateModelChangeCount).toBe(2, 'Start Date 2');
+    expect(fixture.componentInstance.endDateModelChangeCount).toBe(3, 'End date 2');
```

Comment:
Sorry about that. I put in the descriptions so I could tell what was going on and forgot to change them. I have updated them.
End date is on its third change because when start date is set the second time the end date is set to null at that same time.
If you run this test case against the current codebase, you will see that both start and end date will have been set after each call to select, totaling for times each at the end.